src/switchcraft/services/history_service.py

```
@@ -0,0 +1,60 @@
+        dir_path.mkdir(parents=True, exist_ok=True)
+        return dir_path / "history.json"
+
+    def get_history(self):
+        """Get all history items, sorted by date (newest first)."""
+        if not self.history_file.exists():
+            return []
+        try:
+            with open(self.history_file, 'r', encoding='utf-8') as f:
+                 data = json.load(f)
+                 return sorted(data, key=lambda x: x.get('timestamp', ''), reverse=True)
+        except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)